In [70]:
print("eda")

eda


In [71]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns


In [72]:
df = pd.read_csv("../Data/SalesData.csv")
df.head()

,Order_ID,Date,Customer_ID,Product_Category,Product_Name,Quantity,Unit_Price_INR,Total_Sales_INR,Payment_Method,Delivery_Status,Review_Rating,Review_Text,State,Country
0,ORD100000,20-01-2025,CUST2796,Home & Kitchen,Cookware Set,2,25574.41,51148.82,Credit Card,Returned,1,Waste of money,Sikkim,India
1,ORD100001,28-08-2025,CUST9669,Beauty,Hair Dryer,1,19361.41,19361.41,Debit Card,Returned,5,Excellent product!,Telangana,India
2,ORD100002,27-02-2025,CUST5808,Electronics,Tablet,3,38476.22,115428.66,Cash on Delivery,Delivered,3,Fair deal,Nagaland,India
3,ORD100003,24-02-2025,CUST5889,Electronics,Headphones,5,38145.72,190728.60,Credit Card,Delivered,5,Highly recommend!,Assam,India
4,ORD100004,15-06-2025,CUST9005,Clothing,Saree,5,45940.98,229704.90,UPI,Delivered,5,Highly recommend!,Odisha,India


In [73]:
for i in df.columns:
    print(i)
    print(df[i].nunique())

Order_ID
15000
Date
365
Customer_ID
7259
Product_Category
5
Product_Name
25
Quantity
5
Unit_Price_INR
14981
Total_Sales_INR
14988
Payment_Method
4
Delivery_Status
3
Review_Rating
5
Review_Text
25
State
28
Country
1


In [74]:
df['Product_Category'].value_counts()

Product_Category
Electronics       3036
Books             3035
Clothing          3022
Beauty            2997
Home & Kitchen    2910
Name: count, dtype: int64

In [75]:
df['Product_Name'].value_counts()

Product_Name
Children's Book     636
Perfume             629
Science Textbook    624
T-Shirt             623
Laptop              618
Tablet              616
Smartwatch          615
Air Fryer           614
Headphones          610
Jeans               604
Jacket              601
Hair Dryer          599
Fiction Novel       599
Saree               598
Comics              597
Sneakers            596
Face Cream          596
Lipstick            590
Shampoo             583
Mixer Grinder       582
Biography           579
Smartphone          577
Cookware Set        577
Water Purifier      570
Vacuum Cleaner      567
Name: count, dtype: int64

In [76]:
df.groupby("Product_Category")["Product_Name"].nunique()

Product_Category
Beauty            5
Books             5
Clothing          5
Electronics       5
Home & Kitchen    5
Name: Product_Name, dtype: int64

In [77]:
for category, group in df.groupby("Product_Category"):
    print(f"Category: {category}")
    print(group["Product_Name"].unique())

Category: Beauty
['Hair Dryer' 'Face Cream' 'Lipstick' 'Shampoo' 'Perfume']
Category: Books
['Comics' "Children's Book" 'Biography' 'Fiction Novel' 'Science Textbook']
Category: Clothing
['Saree' 'Jeans' 'Sneakers' 'T-Shirt' 'Jacket']
Category: Electronics
['Tablet' 'Headphones' 'Smartphone' 'Laptop' 'Smartwatch']
Category: Home & Kitchen
['Cookware Set' 'Vacuum Cleaner' 'Air Fryer' 'Mixer Grinder'
 'Water Purifier']


## Product Category Summary

The following observations were made from the product catalog:

- The platform offers **5 product categories**: Electronics, Clothing, Books, Beauty, and Home & Kitchen.
- Each product category contains **5 unique products**, resulting in a total of **25 products** across the platform.
- Customer purchases are **evenly distributed** across all products, with each product receiving approximately **570–636 orders**.
- No single product or category significantly outperformed the others, indicating a balanced customer purchasing pattern.

### Conclusion

The product catalog is well-structured, with an equal number of products in each category. Additionally, the nearly uniform purchase distribution suggests that customer demand is balanced across the entire product portfolio, making the dataset suitable for unbiased comparative analysis across different product categories.

In [78]:
df['Quantity'].value_counts()

Quantity
1    3114
5    2990
4    2985
2    2967
3    2944
Name: count, dtype: int64

In [79]:
df['Review_Rating'].value_counts()

Review_Rating
5    3128
4    3022
3    3013
2    2998
1    2839
Name: count, dtype: int64

In [80]:
df["Feedback"] = df["Review_Rating"].map({
    1: "Negative",
    2: "Negative",
    3: "Neutral",
    4: "Positive",
    5: "Positive"
})

In [81]:
df['Feedback'].value_counts()

Feedback
Positive    6150
Negative    5837
Neutral     3013
Name: count, dtype: int64

In [82]:
df['Review_Text'].value_counts()

Review_Text
Satisfied with the product    658
Excellent product!            653
Fantastic quality!            641
Highly recommend!             637
Could be better               632
Poor packaging                626
Worth every rupee!            615
Good quality                  613
Average                       611
Pretty decent                 609
Quality could improve         607
Terrible experience           601
Not satisfied                 600
Not bad                       597
Would buy again               594
Okay product                  593
Product not as expected       592
Very bad product              586
Loved it!                     582
Fair deal                     580
Waste of money                573
Disappointed                  573
Value for money               548
Defective item                543
Never buying again            536
Name: count, dtype: int64

In [83]:
for feedback, group in df.groupby("Feedback"):
    print(f"\n{'='*50}")
    print(f"{feedback}")
    print(f"{'='*50}")
    print(group["Review_Text"].unique())


Negative
['Waste of money' 'Poor packaging' 'Terrible experience'
 'Product not as expected' 'Never buying again' 'Very bad product'
 'Quality could improve' 'Disappointed' 'Defective item' 'Not satisfied']

Neutral
['Fair deal' 'Okay product' 'Average' 'Could be better' 'Not bad']

Positive
['Excellent product!' 'Highly recommend!' 'Fantastic quality!'
 'Worth every rupee!' 'Satisfied with the product' 'Good quality'
 'Would buy again' 'Loved it!' 'Pretty decent' 'Value for money']


## Customer Review & Feedback Summary

The following observations were made from the customer review data:

- Customers rated products on a **5-point review rating scale**, ranging from **1 (lowest)** to **5 (highest)**.
- The review ratings were categorized into **three feedback groups**: **Positive**, **Neutral**, and **Negative**.
- A total of **6,150 reviews** were classified as **Positive**, **3,013 reviews** as **Neutral**, and **5,837 reviews** as **Negative**.
- Each feedback category consists of meaningful customer review texts, allowing customer opinions to be analyzed more effectively.
- Positive feedback includes comments such as *"Excellent product!"*, *"Highly recommend!"*, and *"Value for money"*, while negative feedback highlights issues like *"Poor packaging"*, *"Waste of money"*, and *"Defective item"*.

### Conclusion

The customer reviews were successfully transformed into three meaningful feedback categories—**Positive**, **Neutral**, and **Negative**. This sentiment grouping simplifies customer opinion analysis and provides a strong foundation for identifying satisfaction trends, evaluating product performance, and generating actionable business insights.

In [84]:
df['Delivery_Status'].value_counts()

Delivery_Status
Delivered    5075
Pending      5044
Returned     4881
Name: count, dtype: int64

In [85]:
df['Payment_Method'].value_counts()


Payment_Method
Cash on Delivery    3827
Credit Card         3800
Debit Card          3727
UPI                 3646
Name: count, dtype: int64

In [86]:
df['State'].value_counts()


State
Sikkim               596
Rajasthan            568
Tamil Nadu           567
Meghalaya            559
Chhattisgarh         556
Manipur              555
Assam                553
Uttar Pradesh        551
Madhya Pradesh       550
Bihar                549
Jharkhand            547
Nagaland             545
Telangana            544
Odisha               542
Tripura              540
Karnataka            539
Maharashtra          531
West Bengal          528
Punjab               527
Goa                  522
Arunachal Pradesh    519
Haryana              518
Kerala               510
Mizoram              505
Gujarat              504
Uttarakhand          503
Himachal Pradesh     492
Andhra Pradesh       480
Name: count, dtype: int64

In [89]:
df.groupby("Review_Text")["Feedback"].value_counts().sort_values(ascending=False)

Review_Text                 Feedback
Satisfied with the product  Positive    658
Excellent product!          Positive    653
Fantastic quality!          Positive    641
Highly recommend!           Positive    637
Could be better             Neutral     632
Poor packaging              Negative    626
Worth every rupee!          Positive    615
Good quality                Positive    613
Average                     Neutral     611
Pretty decent               Positive    609
Quality could improve       Negative    607
Terrible experience         Negative    601
Not satisfied               Negative    600
Not bad                     Neutral     597
Would buy again             Positive    594
Okay product                Neutral     593
Product not as expected     Negative    592
Very bad product            Negative    586
Loved it!                   Positive    582
Fair deal                   Neutral     580
Disappointed                Negative    573
Waste of money              Negative   

In [90]:
df[df["Delivery_Status"] == "Returned"].groupby("State").size().sort_values(ascending=False)


State
Assam                201
Uttar Pradesh        196
Sikkim               193
Tamil Nadu           191
Rajasthan            189
Manipur              188
Meghalaya            187
Nagaland             187
Bihar                186
Chhattisgarh         184
Karnataka            182
West Bengal          180
Telangana            179
Tripura              178
Punjab               177
Madhya Pradesh       171
Goa                  171
Odisha               169
Mizoram              166
Jharkhand            163
Kerala               162
Arunachal Pradesh    161
Gujarat              159
Haryana              158
Uttarakhand          158
Himachal Pradesh     155
Maharashtra          151
Andhra Pradesh       139
dtype: int64

In [93]:
feedback = pd.crosstab(df["State"], df["Feedback"], normalize="index") * 100
feedback.sort_values("Positive", ascending=False)

Feedback,Negative,Neutral,Positive
State,,,
Jharkhand,36.745887,18.098720,45.155393
Gujarat,38.095238,17.857143,44.047619
Andhra Pradesh,37.083333,18.958333,43.958333
Bihar,38.433515,17.668488,43.897996
Tripura,35.185185,20.925926,43.888889
West Bengal,38.446970,18.371212,43.181818
Haryana,35.135135,22.007722,42.857143
Manipur,39.099099,18.198198,42.702703
Goa,40.996169,16.858238,42.145594


In [94]:
feedback.sort_values("Negative", ascending=False)

Feedback,Negative,Neutral,Positive
State,,,
Chhattisgarh,44.064748,18.345324,37.589928
Arunachal Pradesh,41.233141,19.075145,39.691715
Goa,40.996169,16.858238,42.145594
Telangana,40.441176,18.382353,41.176471
Meghalaya,40.429338,20.751342,38.819320
Uttar Pradesh,40.290381,21.415608,38.294011
Sikkim,39.765101,20.469799,39.765101
Uttarakhand,39.761431,21.272366,38.966203
Maharashtra,39.736347,18.267420,41.996234


In [95]:
(
    df[df["Feedback"] == "Negative"]
    .groupby("Product_Name")
    .size()
    .sort_values(ascending=False)
)

Product_Name
Hair Dryer          260
Saree               254
Jeans               251
Perfume             250
Science Textbook    245
Fiction Novel       244
Smartphone          244
Children's Book     243
T-Shirt             242
Jacket              237
Sneakers            237
Smartwatch          236
Air Fryer           235
Cookware Set        235
Headphones          232
Laptop              227
Shampoo             227
Tablet              227
Biography           223
Vacuum Cleaner      222
Comics              222
Lipstick            221
Face Cream          214
Mixer Grinder       214
Water Purifier      195
dtype: int64

In [96]:
(
    df.groupby("Product_Category")["Total_Sales_INR"]
    .sum()
    .sort_values(ascending=False)
)

Product_Category
Beauty            2.274896e+08
Electronics       2.265649e+08
Books             2.249992e+08
Clothing          2.224093e+08
Home & Kitchen    2.166987e+08
Name: Total_Sales_INR, dtype: float64

In [98]:
(
    df.groupby("Product_Category")["Total_Sales_INR"]
    .sum()
    .sort_values(ascending=False)
)

Product_Category
Beauty            2.274896e+08
Electronics       2.265649e+08
Books             2.249992e+08
Clothing          2.224093e+08
Home & Kitchen    2.166987e+08
Name: Total_Sales_INR, dtype: float64

In [99]:
(
    df.groupby("Product_Name")["Total_Sales_INR"]
    .sum()
    .sort_values(ascending=False)
    .head(10)
)

Product_Name
Lipstick           48159020.15
Children's Book    48145664.52
Headphones         48044405.75
Hair Dryer         47428844.81
Perfume            46284732.62
Laptop             46173868.15
Tablet             46170022.15
Sneakers           46057295.93
Air Fryer          45919707.91
Jeans              45232819.47
Name: Total_Sales_INR, dtype: float64

In [100]:
(
    df.groupby("State")["Total_Sales_INR"]
    .sum()
    .sort_values(ascending=False)
)

State
Sikkim               43113469.51
Rajasthan            42906175.08
Chhattisgarh         42857545.27
Meghalaya            42773152.96
Tamil Nadu           41967968.99
Uttar Pradesh        41690917.07
Bihar                41669240.44
West Bengal          41195932.45
Tripura              41103376.81
Odisha               40924381.38
Goa                  40488243.03
Madhya Pradesh       39961650.61
Manipur              39934040.52
Punjab               39640897.69
Jharkhand            39606674.07
Arunachal Pradesh    39348777.94
Kerala               39279218.92
Haryana              39171925.41
Telangana            39077114.06
Assam                38739460.17
Maharashtra          38675511.58
Nagaland             38670484.69
Mizoram              38641789.63
Karnataka            38444302.71
Himachal Pradesh     38182171.87
Uttarakhand          37633617.48
Andhra Pradesh       36237577.25
Gujarat              36226185.91
Name: Total_Sales_INR, dtype: float64

In [101]:
(
    df.groupby("Payment_Method")["Total_Sales_INR"]
    .sum()
    .sort_values(ascending=False)
)

Payment_Method
Credit Card         2.868892e+08
Cash on Delivery    2.865114e+08
Debit Card          2.730366e+08
UPI                 2.717245e+08
Name: Total_Sales_INR, dtype: float64

# Business Insights

## 1. State-wise Returned Orders

- **Assam (201)** recorded the highest number of returned orders, followed by **Uttar Pradesh (196)** and **Sikkim (193)**.
- **Andhra Pradesh (139)** had the lowest number of returned orders among all states.
- The difference in return counts across states is relatively small, indicating that product returns are fairly consistent throughout the regions.

**Business Insight:**  
States with higher return volumes, particularly Assam and Uttar Pradesh, should be investigated to identify whether returns are driven by product quality, delivery issues, incorrect orders, or customer expectations.

---

## 2. State-wise Customer Feedback Analysis

### States with the Highest Positive Feedback

- **Jharkhand** recorded the highest percentage of positive customer feedback (**45.16%**).
- **Gujarat (44.05%)**, **Andhra Pradesh (43.96%)**, **Bihar (43.90%)**, and **Tripura (43.89%)** also demonstrated high customer satisfaction.
- These states consistently showed higher customer satisfaction compared to the overall average.

### States with the Highest Negative Feedback

- **Chhattisgarh** recorded the highest percentage of negative feedback (**44.06%**).
- Other states with relatively higher negative feedback include **Arunachal Pradesh**, **Goa**, **Telangana**, and **Meghalaya**.

**Business Insight:**  
Customer satisfaction varies across states. States with higher negative feedback should be prioritized for quality improvement initiatives, customer service enhancements, and operational reviews to improve the overall customer experience.

---

## 3. Products Receiving the Most Negative Feedback

- **Hair Dryer (260 reviews)** received the highest number of negative customer feedback.
- Other products with high negative feedback include:
  - Saree (254)
  - Jeans (251)
  - Perfume (250)
  - Science Textbook (245)
  - Smartphone (244)
  - Fiction Novel (244)

-  **Water Purifier (195)** received the lowest number of negative reviews among all products.

**Business Insight:**  
Products receiving consistently high negative feedback should undergo detailed quality assessment. Common customer complaints should be analyzed to determine whether improvements are required in manufacturing, packaging, or product specifications.

---

## 4. Category-wise Revenue Analysis

Total Revenue Generated:

| Category | Revenue (Approx.) |
|-----------|------------------:|
| Beauty | ₹227.49 Million |
| Electronics | ₹226.56 Million |
| Books | ₹225.00 Million |
| Clothing | ₹222.41 Million |
| Home & Kitchen | ₹216.70 Million |

- **Beauty** generated the highest revenue.
- **Home & Kitchen** generated comparatively lower revenue than other categories.
- Revenue across categories remains relatively balanced, indicating healthy customer demand throughout the product catalog.

**Business Insight:**  
No single category dominates overall sales, suggesting a diversified product portfolio with consistent customer demand.

---

## 5. Top Revenue Generating Products

Top-performing products include:

1. Lipstick
2. Children's Book
3. Headphones
4. Hair Dryer
5. Perfume
6. Laptop
7. Tablet
8. Sneakers
9. Air Fryer
10. Jeans

-  **Lipstick** generated the highest revenue (approximately **₹48.16 Million**).
-  Several products from different categories appear among the top revenue generators, indicating balanced product performance.

**Business Insight:**  
These products should be prioritized for inventory planning, promotional campaigns, and customer retention strategies due to their significant contribution to total revenue.

---

## 6. State-wise Revenue Analysis

- **Sikkim** generated the highest total revenue (approximately **₹43.11 Million**).
- Rajasthan, Chhattisgarh, Meghalaya, and Tamil Nadu also contributed significantly to overall sales.
- Gujarat and Andhra Pradesh generated comparatively lower total revenue.

**Business Insight:**  
High-performing states provide opportunities for additional marketing investment, while lower-performing states may benefit from localized promotional campaigns and customer acquisition strategies.

---

## 7. Revenue by Payment Method

Revenue distribution by payment method:

- Credit Card — Highest Revenue
- Cash on Delivery — Very close second
- Debit Card
- UPI

- Credit Card transactions generated the highest overall revenue.
- Cash on Delivery remains a highly preferred payment option among customers.
- Revenue generated through all payment methods is relatively balanced.

**Business Insight:**  
Supporting multiple payment options is essential, as customers continue to utilize a variety of payment methods. Promotional offers on digital payments may encourage greater adoption of cashless transactions.

---

# Overall Conclusion

The e-commerce platform demonstrates a **well-balanced business performance** across product categories, products, payment methods, and geographical regions.

Key findings include:

- Beauty products generated the highest overall revenue.
- Lipstick emerged as the highest revenue-generating product.
- Assam recorded the highest number of returned orders.
- Jharkhand achieved the highest customer satisfaction, while Chhattisgarh showed the highest proportion of negative feedback.
- Hair Dryer received the largest number of negative customer reviews, indicating potential opportunities for product improvement.
- Revenue is evenly distributed across categories and payment methods, reducing dependency on any single product segment.

Overall, the analysis suggests that while the platform maintains strong and diversified sales performance, targeted improvements in product quality, packaging, and customer experience in specific states and products could further enhance customer satisfaction and reduce return rates.